src/lift.jl

```diff
@@ -0,0 +1,59 @@
+"""
+    lift(f, x)
+
+Return `f(x)` unless `x isa Union{Nothing,Missing}`, in which case return `missing`.
+
+This is particularly useful when handling values from `Arrow.Table`, whose null values
+may present as either `missing` or `nothing` depending on how the table itself was
+originally constructed.
+
+See also: [`lift_type`](@ref)
+"""
+lift(f, x) = f(x)
+lift(::Any, ::Union{Nothing,Missing}) = missing
+
+"""
+    lift(f)
+
+Returns a curried function, `x -> lift(f,x)`
+"""
+lift(f) = Base.Fix1(lift, f)
+
+
+"""
+    construct(T::Type, x)
+
+Construct `T(x)` unless `x` is of type `T`, in which case return `x` itself. Useful in
+conjunction with the [`lift`](@ref) function for types which don't have a constructor which
+accepts instances of itself (e.g. `T(::T)`).
+
+## Examples
+
+```jldoctest
+julia> using Legolas: construct
+
+julia> construct(Float64, 1)
+1.0
+
+julia> Some(Some(1))
+Some(Some(1))
+
+julia> construct(Some, Some(1))
+Some(1)
+```
+
+Use the curried form when using `lift`:
+
+```jldoctest
+julia> using Legolas: lift, construct
+
+julia> lift(Some, Some(1))
+Some(Some(1))
+
+julia> lift(construct(Some), Some(1))
+Some(1)
+```
+"""
+construct(T::Type, x) = T(x)
+construct(::Type{T}, x::T) where T = x
+construct(T::Type) = Base.Fix1(construct, T)
```

Comment:
```suggestion
construct(T, x) = T(x)
construct(::Type{T}, x::T) where {T} = x
construct(T) = Base.Fix1(construct, T)
```

IMO there's no need for the `::Type` constraint on `T` for the fallback case

(also `where T` -> `where {T}` for YASGuide purposes)

Reply:
Although the type constraint isn't needed for functionality this does generalize the function such that the docstring is no accurate (`T` may not be a type) and the not using a type for `T` would also result in me no longer wanting to use the name `construct` for the function as construction/constructors are tied to types

Reply:
> generalize the function such that the docstring is no accurate (T may not be a type) 

yeah, my suggestion here IS to generalize the function such that `T` may not be a type (if you wanted could rename to `f` too)

> construction/constructors are tied to types

i see no reason to restrict it to types vs. any generic non-type callable

but i'm also fine moving forward with this PR without generalizing if there's some concern there

(aside: this is such a bikesheddable bite-sized PR @omus , sorry 😂)

Reply:
> this is such a bikesheddable bite-sized PR @omus , sorry 😂

Very true. I think we can just proceed with this PR as it currently exists. If we want to generalize this function further I think the name should change which means we can easily replace this function with something else in the future.